crates/bevy_dev_tools/src/fps_overlay.rs

```diff
@@ -95,35 +97,36 @@ fn setup(mut commands: Commands, overlay_config: Res<FpsOverlayConfig>) {
             },
             GlobalZIndex(FPS_OVERLAY_ZINDEX),
         ))
-        .with_children(|c| {
-            c.spawn((
-                TextBundle::from_sections([
-                    TextSection::new("FPS: ", overlay_config.text_config.clone()),
-                    TextSection::from_style(overlay_config.text_config.clone()),
-                ]),
-                FpsText,
-            ));
-        });
+        .spawn_text_block::<TextNEW>([
```

Comment:
> Given that one of the motivations behind this refactor is to make text easily representable as raw components, I find it a bit concerning that we're building new special case abstractions on top. I would prefer it if we removed this and used the normal spawn APIs to create the components directly.

The raw API makes spawning big lists of spans much more painful. I've used `spawn_text_block` multiple times while migrating examples and it is very helpful. If you want to aim at a different API I suggest grepping `spawn_text_block` in this branch and trying to migrate some of the heavy-user-examples to your vision of raw spawning (one example has 22 or more spans). If you can make it not painful, then I'll accept suggestions without resistance.

> I think this should be replaced with the original impl from this PR (directly spawning the components), and the helper method should be removed.

Doing that will make this PR a regression for existing users, rather than a step forward (or at least sideways). I would be fine with removing these ergonomic workarounds *after* viable alternatives are available.

Reply:
I've just pushed a commit that removes spawn_text_block. In many places it actually improves legibility / ergonomics because it doesn't force style to be specified in non-default cases. It also really helps legibility in cases where we need to add additional components to the root node.

I know you probably still disagree, but I'm making an executive decision for the reasons I've stated above (and the new ones discovered during the port).

For the duration of this PR, lets stick to normal spawning. We can pick up the ergonomics question later. But imo those changes were in general a lateral move ergonomically and they came at the cost of obfuscation.

Reply:
Yeah I found that just updating the stress test examples was really painful, I think without something that just leaves users with the burden of having to invent their own adhoc builders and these changes aren't immediately easy to understand anyway.

`spawn_text_block` isn't a great solution I realised though, it doesn't allow users to add marker components for instance.